### ISSUE TITLE
The "debian" folder of a newly added platform driver is ignored to be uploaded.

### ISSUE THREAD

**Issue by Komil-yc** [Azure/sonic-buildimage#7683]
#### Description

    There is a pattern "platform/**/debian/*" described in .gitignore.
That causes GIT to ignore all files in debian/ used to build Debian package. 
A newly added platform will miss debian/ folder to finish their building. 
The change is introduced by below PR:
      https://github.com/Azure/sonic-buildimage/pull/4803
    
Is that an expected behavior to ignore debian/ folder automatically and
should we upload them by another way like "git add -f" ?
#### Steps to reproduce the issue:
1.
2.
3.

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by anshuv-mfst** [comment]
Hi @jleveque - could you please take a look, thanks.

**Comment by jleveque** [comment]
I agree that this ignore is a bit severe, and could cause vendors to miss adding their debian directory. I will open a PR to remove this line. However, I think each vendor should be responsible for providing one or more `.gitignore` files to properly ignore generated files in their platform directories.

**Comment by bluecmd** [comment]
I created a PR to mitigate this issue as I lost a few hours today because of files being hidden from `git status`; https://github.com/sonic-net/sonic-buildimage/pull/14565
